apps/prairielearn/elements/pl-string-input/pl-string-input.py

```diff
@@ -107,7 +107,7 @@ def render(element_html: str, data: pl.QuestionData) -> str:
                     "Leading and trailing spaces will be left as part of your answer."
                 )
             case _:
-                raise Exception("Should never reach here.")
+                assert_never("Should never reach here.")
```

Comment:
I think you want this instead:

```suggestion
                assert_never(space_hint_pair)
```

Reply:
Ah, thanks. I read the docs on what assert_never takes in, I thought it was for something else.